var/spack/repos/builtin/packages/hicops/package.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Hicops(CMakePackage):
+    """HiCOPS is a software framework for accelerating database peptide search
+    workflows on supercomputers. HiCOPS provided algorithm-independent
+    parallelizations and optimizations can be extended into new HPC database search
+    algorithms or scalably accelerate the existing ones.
+    """
+
+    homepage = "https://hicops.github.io/index"
+    git = "https://github.com/hicops/hicops.git"
+    maintainers = ['pcdslab', 'mhaseeb123', 'nessiecancode']
+
+    version('release', branch='release')
+    version('develop', branch='develop')
+
+    # Build Options
+    variant('mpi', default=True,
+            description='Enable MPI support.')
+    variant('timemory', default=False,
+            description='Enable timemory interface. Requires timemory '
+                        'installation.')
+    variant('mpip', default=False,
+            description='Enables the MPIP data_tracker via Timemory. '
+                        'Requires timemory installation.')
+    variant('tailfit', default=True,
+            description='Use the tailfit method instead of Gumbelfit '
+                        'for e-value computation.')
+    variant('progress', default=True,
+            description='Display HiCOPS progress marks.')
+    variant('seqlen', default='60',
+            description='Allowed maximum peptide sequence length.')
+    variant('qalen', default='100',
+            description='Maximum number of top K peaks to keep when '
+                        'spectrum preprocess.')
+    variant('qchunk', default='10000',
+            description='Max size of each batch extracted from the '
+                        'dataset.')
+    variant('hyperscore', default='100',
+            description='Maximum allowed hyperscore computed.')
+    variant('shdpeaks', default='80',
+            description='Maximum shared b- or y-ions allowed.')
```

Comment:
These are all missing possible values, can we add them here too? If not any string is accepted as a possible value.

Reply:
@alalazo is the use of a range acceptable, as some have a number of values acceptable example seqlen can be set anywhere between 7 to 60

and some are tunable variants with no true range just recommendation, qchunk, hyperscore, shdpeaks, qalen, can I just sanitize the value to just be numerical?

Reply:
@mhaseeb123 do you have recommended values that should be used for your application that are reasonable for someone to set it outside of the default? 

QALEN
QCHUNK
MAX_HYPERSCORE
MAX_SEQ_LEN
SHDPEAKS

Reply:
> can I just sanitize the value to just be numerical?

Yes, you can say for instance `values=int` to validate the values are `int`.

Reply:
@alalazo okay, I went ahead and added values=int to the variants that were missing values

Reply:
Hi @NessieCanCode , yeah I think we can do some santization. Here is a range of appropriate values that I think should be for the above variables. Now I am thinking, I should move some of these variables to runtime instead of compile time, maybe in the next release.

50 <= QALEN <= 500
5000 <= QCHUNK <= 50000
20 <= MAX_HYPERSCORE <= 200
MAX_SEQ_LEN <= 60
MIN_SEQ_LEN >= 6
MAX_SHDPEAKS <= 20 (in general it should be <= QALEN)

Please make sure the variant `shdpeaks` translates to the variable `MAX_SHDPEAKS` instead of `SHDPEAKS` as it does not exist. Might be a good idea to check if other variants point to correct CMake variables. Please let me know if there is anything. Thank you.